libs/net/services/Actions/Managers/DataSourceIngestManager`.cs

```diff
@@ -120,7 +120,7 @@ public virtual bool VerifyDataSource()
     public virtual bool VerifySchedule(DateTime date, ScheduleModel schedule)
     {
         if (!this.DataSource.IsEnabled || !this.DataSource.DataSourceSchedules.Any(s => s.Schedule?.IsEnabled == true)) return false;
-        return VerifyDelay(date, schedule) &&
+        return VerifyDelay(DateTime.UtcNow, schedule) &&
```

Comment:
As VerifyDelay() always needs to use the time in UTC (as we discussed, and I believe you agreed) this was my way of implementing this. I considered getting the DateTime.UtcNow inside VerifyDelay(), but that would make the date parameter redundant and would require a change to the VerifyDelay() definition in IDataSourceIngestManager. I thought this was easier. 

The bug I saw was introduced because of the call to GetSourceDateTime() in the line:

VerifySchedule(GetSourceDateTime(DateTime.UtcNow), s.Schedule!)

Which isn't required for VerfiyDelay(). The version in dev returns:

return next <= date.ToUniversalTime();

Which I believe is meant to turn the date value, which was set to the data source's timezone, back into universal time. This doesn't work. It seems to turn the Kind of the DateTime object to UTC, but leaves the actual time as that of the data source's timezone. E.g.

It transforms 13:43:09 EST to 13:43:09 UTC

This results in comparing next (the LastRanOn value in UTC plus schedule.DelayMS) with a time that's approximately 7 hours in the future.

Reply:
We did discuss, but reviewing the code shows issues with your implementation.  The function is determining if the configured delay has occurred based on a comparison between `LastRanOn` and the provided `date`.  The function is not comparing `DateTime.UtcNow` to `LastRanOn`.

Reply:
The provided date is DateTime.UtcNow in the data source's time zone.

VerifySchedule(GetSourceDateTime(DateTime.UtcNow), s.Schedule!)

As the original implementation just turns the value of date back into UTC why not just pass the value of DateTime.UtcNow?

Reply:
The function is comparing a specific `date` to `schedule.LastRanOn`.  This is important, as it's a testable and reusable function to compare two specified values.  Your code invalidates the function.  If there is an issue with the timezone related comparison between `date` and `schedule.LastRanOn` then that should be addressed in some way.  You should not change `date` to an unknown, uncontrollable value `DateTime.UtcNow`.

Reply:
How would you address the fact that date.ToUniversalTime(); doesn't do what we want?

Reply:
I regrettably don't fully understand why the error is occurring.  I'm assuming because the `date` and the `schedule.LastRanOn` are not using the same timezone.  If this is the case, then convert accordingly.

Reply:
That's correct. I found an ugly, and probably slow, way to do it. I may also have spotted another solution, so I can follow that lead.

Reply:
Wouldn't the `schedule.LastRanOn.ToTimeZone(GetTimeZone())` do what you need?

Reply:
Yes.